queue_processors/queue_processor/queueproc_utils/variable_utils.py

```diff
@@ -47,15 +47,8 @@ def copy_variable(variable):
         Return a temporary copy (unsaved) of the variable,
         which can be modified and then saved without modifying the original.
         """
-        return InstrumentJoin(name=variable.name,
-                              value=variable.value,
-                              is_advanced=variable.is_advanced,
-                              type=variable.type,
-                              help_text=variable.help_text,
-                              instrument=variable.instrument,
-                              experiment_reference=variable.experiment_reference,
-                              start_run=variable.start_run,
-                              tracks_script=variable.tracks_script)
+        variable.pk = None  # Creates a copy by changing primary key to None
+        return variable
 
     @staticmethod
     def get_type_string(value):
```

Comment:
NIT (as not part of the code you have edited) Should we log if an unknown type is passed when returning text for unknown type inside `get_type_string?`

Reply:
We control all these types via the database enum so I'm not too concerned plus the default fall through is to text which should handle any types we do not expect